docs/concepts/standard-library/gnopher-hole.md

```diff
@@ -0,0 +1,105 @@
+   name and signature, i.e. `crypto/sha256/sha256.go`.
+3. When the two above are present and valid, the native binding can be created
+   by executing the code generator: either by executing `go generate` from the
+   `stdlibs` directory, or run `make generate` from the `gnovm` directory.
+   This generates the `native.go` file available in the `stdlibs` directory,
+   which provides the binding itself to then be used by the GnoVM.
+
+The code generator in question is available in the `misc/genstd` directory.
+There are some quirks and features that must be kept in mind when writing native
+bindings, which are the following:
+
+- Unexported functions (i.e. `func sum256(b []byte)`) must have their
+  Go counterpart prefixed with `X_` in order to make the functions exported (i.e.
+  `func X_sum256(b []byte)`).
+- The Go function declaration may specify as the first argument
+  `m *gno.Machine`, where `gno` is an import for
+  `github.com/gnolang/gno/gnovm/pkg/gnolang`. This gives the function access to
+  the Virtual Machine state, and is used by functions like `std.AssertOriginCall()`.
+- The Go function may change the type of any parameter or result to
+  `gno.TypedValue`, where `gno` is an import for the above import path. This
+  means that the `native.go` generated code will not attempt to automatically
+  convert the Gno value into the Go value, and can be useful for unsupported
+  conversions like interface values.
+- A small set of named types are "linked" between their Gno version and Go
+  counterpart. For instance, `std.Address` in Gno is
+  `(".../tm2/pkg/crypto").Bech32Address` in Go. A list of these can be found in
+  `misc/genstd/mapping.go`.
+- Not all type literals are currently supported when converting from their Gno
+  version to their Go counterpart, i.e. `struct` and `map` literals. If you intend to use these,
+  modify the code generator to support them.
+- The code generator does not inspect any imported packages from the Go native code
+  to determine the default package identifier (i.e. the `package` clause).
+  For example, if a package is in `foo/bar`, but declares `package xyz`, when importing
+  foo/bar the generator will assume the name to be `bar` instead of `xyz`.
+  You can add an identifier to the import to fix this and use the identifier
+  you want/need, such as `import gno "github.com/gnolang/gno/gnovm/pkg/gnolang"`.
+
+## Adding new standard libraries
+
+New standard libraries may be added by simply creating a new directory (whose
+path relative to the `stdlibs` directory will be the import path used in Gno
+programs). Following that, the suggested approach for adding a Go standard
+library is to copy the original files from the Go source tree, and renaming their
+extensions from `.go` to `.gno`.
+
+> As a small aid, this bash one-liner can be useful to convert all the file
+> extensions:
+>
+> ```sh
+> for i in *.go; do mv $i "$(echo $i | sed 's/\.go$/.gno/')"; done
+> ```
```

Comment:
Please move this section into an [admonition](https://docusaurus.io/docs/2.x/markdown-features/admonitions)

Reply:
[3eaacce](https://github.com/gnolang/gno/pull/1468/commits/3eaacce8e326f807b8419ab5c46bf491b82f7fb7)